opnode/l2/sync_start.go

```diff
@@ -59,7 +59,7 @@ func FindSyncStart(ctx context.Context, reference SyncReference, genesis *Genesi
 	// Search back: linear walk back from engine head. Should only be as deep as the reorg.
 	for refL2.Number > 0 {
 		// remember the canonical L1 block that builds on top of the L1 source block of the L2 parent block.
-		nextRefL1 = refL1
+		nextRefL1 = currentL1
```

Comment:
This was a bug, found with new tests, fixed now. When traversing back, remember the actual L1 hash as canonical, not the L1 hash from the engine block